### DISCUSSION TITLE
PTS_MODE versus manual/scripted input data

### DISCUSSION THREAD

**Discussion by evalieungh** [NorESMhub/noresm-land-sites-platform#149] [General]
Recently, @ka7eh has been developing a new branch where PTS_MODE, i.e. the `subset_data` script etc., is used to create input data. This has the advantages of:

- making it easier to add new sites for users
- making it easier to add new sites for developers
- keeping developments close to NCAR's coding solutions

But there are also downsides:

- restart simulations are not supported ([see docs](https://escomp.github.io/ctsm-docs/versions/release-clm5.0/html/users_guide/running-single-points/running-pts_mode-configurations.html))
- maybe less flexibility in other ways?
- strategically, we risk downvaluing @huitang-earth and @lasseke's work on creating the scripts and workflow in the existing [input repo](https://github.com/NorESMhub/noresm-lsp-input)

One solution is to use PTS_MODE as the main branch, and have another `legacy` branch for the existing method. I guess it's possible to swap the two as well if someone has capacity to develop better input data and implement restart options in the LSP in the future. For now, in case we get lots of new users, I support the suggestion to use PTS_MODE as the default because it makes users more self-sufficient.

**Comment by lasseke** [comment]
A quick additional explanation of why we decided to also include `subset_data` for the first release: the journal we aim for requires that software described in "technical advances" articles is not limited to specific use cases. The proper way of adding custom single-point simulations requires many changes in the XML files we describe in the docs of the `noresm-lsp-input` repo. However, as this is technically demanding, this would require significantly more maintenance work, which we cannot guarantee to provide (short-term contracts, etc.).

As we did not have enough time to implement a user-friendly way for using restart files or to properly manipulate the input data and only use default data either way, we thought that this would be a good short-term solution to provide an easy-access educational version. An additional downside is that users will need to download larger global datasets (~20 GB), which are then shared among different cases.
